commands/curation/curationaudit.go

```diff
@@ -253,11 +271,16 @@ func (ca *CurationAuditCommand) auditTree(tech coreutils.Technology, results map
 		tech:                 tech,
 		parallelRequests:     ca.parallelRequests,
 	}
-	packagesStatusMap := sync.Map{}
+
+	rootNodes := map[string]struct{}{}
+	for _, tree := range fullDependenciesTrees {
+		rootNodes[tree.Id] = struct{}{}
+	}
 	// Fetch status for each node from a flatten graph which, has no duplicate nodes.
-	err = analyzer.fetchNodesStatus(flattenGraph, &packagesStatusMap, rootNode.Id)
-	analyzer.fillGraphRelations(rootNode, &packagesStatusMap,
-		&packagesStatus, "", "", datastructures.MakeSet[string](), true)
+	packagesStatusMap := sync.Map{}
+	err = analyzer.fetchNodesStatus(flattenGraph, &packagesStatusMap, rootNodes)
```

Comment:
if err exists after this line why are we not checking and continue to process until the end?
should we check for error here?

Reply:
fetchNodeStatus does alot of requests and if some are failing, we still want to produce a report and prompt that there was an error. I'll add a comment